geonode/qgis_server/views.py

```diff
@@ -117,33 +119,14 @@ def download_qgs(request, layername):
 
     :return: QGS file.
     """
-
     layer = get_object_or_404(Layer, name=layername)
-    ogc_url = reverse('qgis_server:layer-request',
-                      kwargs={'layername': layername})
-    url = settings.SITEURL + ogc_url.replace("/", "", 1)
-
-    layers = [{
-        'type': 'raster',
-        'display': layername,
-        'driver': 'wms',
-        'crs': 'EPSG:4326',
-        'format': 'image/png',
-        'styles': '',
-        'layers': layer.title,
-        'url': url
-    }]
-
-    json_layers = json.dumps(layers)
-
-    url_server = settings.QGIS_SERVER_URL + \
-        '?SERVICE=PROJECTDEFINITIONS&LAYERS=' + json_layers
-    request = requests.get(url_server)
+    url = qgs_url(layer, internal=True)
+    request = requests.get(url)
```

Comment:
misleading name, this is a response object (from python requests lib).
So, I suggest changing the var name `request` into `result` for example.